core/kazoo_json/src/kz_json.erl

```diff
@@ -245,6 +245,26 @@ are_equal(JObj1, JObj2) ->
 from_list([]) -> new();
 from_list(L) when is_list(L) -> ?JSON_WRAPPER(L).
 
+-spec from_list_recursive(json_proplist()) -> object().
+from_list_recursive([]) -> new();
+from_list_recursive(L)
+  when is_list(L) ->
+    recursive_from_list(L).
+
+-spec recursive_from_list(list()) -> object().
+recursive_from_list([First | _]=List)
+  when is_list(List)
+       andalso is_tuple(First)  ->
+    set_values([{K, recursive_from_list(V)} || {K,V} <- List], new());
+recursive_from_list(X) when is_float(X) -> X;
+recursive_from_list(X) when is_integer(X) -> X;
+recursive_from_list(X) when is_atom(X) -> X;
```

Comment:
A test example: try a nested get in the created jobj for an atom key & the same binary key. That would be interesting to test/document.